### DISCUSSION TITLE
Request timeout

### DISCUSSION THREAD

**Discussion by Borgran** [openai-php/client#223] [Q&A] [answered]
Is there a way to pass increased request timeout parameter? I am running simple request like:

`$response = $client->chat()->create([
				'model' => 'gpt-3.5-turbo',
				'messages' => [
					['role' => 'user', 'content' => $command],
				],
				'max_tokens' => 1500,
				'temperature' => 1,
			]);`

and getting timeouts:

`Symfony\Component\HttpClient\Exception\TimeoutException: Idle timeout reached for "https://api.openai.com/v1/chat/completions". in ./vendor/symfony/http-client/Chunk/ErrorChunk.php:56`

**Comment by Borgran** [comment]
`$client = OpenAI::factory()
->withApiKey(OPENAI_API_KEY)
->withBaseUri('api.openai.com/v1') //default: api.openai.com/v1
->withHttpClient($client = new \GuzzleHttp\Client(['timeout' => 600,])) 
->make();`

**Reply by Ranktotop** [reply]
This gives "Expected parameter of type 'Psr\Http\Client\ClientInterface', 'GuzzleHttp\Client' provided "

**Comment by StandingFuture** [comment]
Would be great to be able to specify request timeout. I have had to switch my app to gpt3.5 since gpt4-turbo keeps timing out.

**Comment by gehrisandro** [comment, marked as answer]
Because this library is HTTP client agnostic, there is no way for us to configure the timeout globally.

But I've added a section to the docs, explaining how to set timeouts: https://github.com/openai-php/client?tab=readme-ov-file#timeout

**Reply by StandingFuture** [reply]
Thank you so much! Seems to be working!
